### ISSUE TITLE
More restrictive dropbox permissions.

### ISSUE THREAD

**Issue by 0e318236-1680-41a8-a989-3c9227637cdf** [mickael-kerjean/nuage#60]
I think the demo is requesting too much permission from Dropbox.

There is the option of giving Nuage only access to its own folder in Dropbox. https://www.dropbox.com/help/security/third-party-apps

This looks like an service that someone can install on their own server, but I think permissions should be isolated so that if codebase, or application is compromised that damage is isolated.

**Comment by mickael-kerjean** [comment]
You're right in a way but unfortunatly you can't have multiple permission type while creating a n API key with Dropbox. As I see it now, the best fix would be to make it work with any permission type coming from Dropbox, after all the demo is just a preconfigure version.

If you have a better idea, feel free to share

**Comment by mickael-kerjean** [comment]
I've been thinking about this a bit more and I'm not willing to change the permission type of the demo as it gives a good overview of some capabilities dropbox don't offer like editing any file with an proper editor, audio player, video player, ....

However I will make sure the dropbox api can be used with multiple permission type so that anyone willing to selfhost can make it work with the security model he wants.

From a security standpoint, the server isn't storing anything at all, credentials are only kept in RAM as they are decrypted from the HTTP only cookie. In the worst case scenario where the server is compromised, one would need to actively look at the cookies in every request coming in. As I've seen it, the only actor that provide folder based permission for an app is dropbox and if such an event occure and is detected, I'd prefer be transparent via communication (I created a ticket about [this specific point](https://github.com/mickael-kerjean/nuage/issues/57))

**Comment by mickael-kerjean** [comment]
I made a few test with Dropbox and it seems to be working fine with both permissions: "App Folder" and "Full Dropbox".

To make it works you can either:
1. create the DROPBOX_CLIENT_ID and APPLICATION_URL
2. edit the [config_server.js](https://github.com/mickael-kerjean/nuage/blob/master/config_server.js)

As dropbox permission mechanism assume permission are set in the marble when you generate the API key, it's hard to make things better. Any suggestion?
